src/shared/service-util.c

```diff
@@ -36,15 +39,19 @@ static int help(const char *program_path, const char *service, const char *descr
 int service_parse_argv(
                 const char *service,
                 const char *description,
+                const BusObjectImplementation* const* bus_objects,
                 int argc, char *argv[]) {
 
         enum {
                 ARG_VERSION = 0x100,
+                ARG_BUS_INTROSPECT,
         };
 
-        static const struct option options[] = {
+        const struct option options[] = {
                 { "help",           no_argument,       NULL, 'h'                },
                 { "version",        no_argument,       NULL, ARG_VERSION        },
+                { bus_objects ? "bus-introspect" : NULL,
+                                    required_argument, NULL, ARG_BUS_INTROSPECT },
```

Comment:
hmm, i'd leave the table constant, but then just output a nice error message if anyone tries it on a service that doesn't have this. mutable option tables are weird...